specification/dartLangSpec.tex

```diff
@@ -16276,38 +16279,40 @@ \subsection{Assignable Expressions}
 \end{grammar}
 
 \LMHash{}%
-An \Index{assignable expression} is either:
-\begin{itemize}
-\item An identifier.
-\item An invocation (possibly conditional) of a getter (\ref{getters})
-  or list access operator on an expression $e$.
-\item An invocation of a getter or list access operator on \SUPER.
-\end{itemize}
-
-\LMHash{}%
-An assignable expression of the form \id{} is evaluated as
-an identifier expression (\ref{identifierReference}).
-
-%An assignable expression of the form \code{$e$.\id($a_1, \ldots,\ a_n$)}
-% is evaluated as a method invocation (\ref{methodInvocation}).
-
-\LMHash{}%
-An assignable expression of the form \code{$e$.\id} or \code{$e$?.\id}
-is evaluated as a property extraction (\ref{propertyExtraction}).
-
-\LMHash{}%
-An assignable expression of the form \code{$e_1$[$e_2$]}
-is evaluated as a method invocation of
-the operator method \code{[]} on $e_1$ with argument $e_2$.
-
-\LMHash{}%
-An assignable expression of the form \code{\SUPER.\id} is evaluated as
-a property extraction.
+\BlindDefineSymbol{e, t}%
+Let $e$ be an \synt{assignableExpression}.
+Let $t$ be a term such that $e$ can be derived from
+\syntax{$t$ <assignableSelector>}.
+We say that $t$ is the
+\IndexCustom{receiver term}{assignable expression!receiver term}
+of $e$.
+When $t$ is an expression, we say that $t$ is the
+\IndexCustom{receiver expression}{assignable expression!receiver expression}
+of $e$.
+
+\commentary{%
+In short, we obtain $t$ by cutting off an assignable selector
+from the end of $e$.
+It is easy to see that every \synt{assignableExpression}
+except one which is simply an \synt{identifier}
+ends in an \synt{assignableSelector}
+(an \synt{unconditionalAssignableSelector}
+is also an \synt{assignableSelector}).
+In other words, every assignable expression except \synt{identifier}
+has a receiver term.
+
+Note that the receiver term $t$ may or may not be an \synt{expression}.
+For example, if $t$ is \SUPER{} then $e$ does not have a receiver expression.
+The semantics of assignments without a receiver expression
+is specified explicitly for each case
+(\ref{assignment}).%
+}
 
 \LMHash{}%
-Evaluation of an assignable expression of the form \code{\SUPER{}[$e_2$]}
-is equivalent to
-evaluation of the method invocation \code{\SUPER.[]($e_2$)}.
+Let $e$ be an assignable expression.
+Assume that $e$ has a receiver expression $t$.
+Evaluation of $t$ proceeds in the same way as
+evaluation of any other expression.
```

Comment:
Yup! This fails if `t` is a prefix, but an identifier _is_ (grammatically) an expression.

Reply:
The text now explicitly states that cases with side conditions other than 'the subterm is an expression' are more specific. So that case is handled by the case that explicitly talks about import prefixes.